### ISSUE TITLE
Can't comment on videos using remote interaction

### ISSUE THREAD

**Issue by drym3r** [Chocobozzz/PeerTube#4799]
**Describe the current behavior**
When visiting a video without logging in our instance and wanting to comment it using remote interaction, it doesn't work.

**Steps to reproduce:**

1. Go to a video, for example [this](https://fediverse.tv/w/ctfm1d6Aa4qxwZ7GNTp1VE).
2. Click the box of "Add comment..." and the box with the text "You are one step away from commenting" opens.
3. Add a username from a remote instance and click to "Remote interact"
4. A page in the remote instance will open with a generic error. On mastodon it shos "You don't have permission to view this page.", in pleroma "Error following account".

**Describe the expected behavior**
I should be able to comment in the video in some way. 

**Additional information**
I've tried with the next instances:
- mastodon.social
- mastodon.art
- barcelona.social

* PeerTube instance: 
  * URL: https://fediverse.tv
  * version: v4.0.0
  * NodeJS version: v12.22.9
  * Ffmpeg version: 4.3.3-0+deb11u1
It uses the docker image `chocobozzz/peertube:v4.0.0-bullseye`.

* Browser name, version and platforms on which you could reproduce the bug: Firefox
* Url that the button gets:
** In mastodon it goes to a url like: `https://mastodon.social/authorize_interaction?uri=https%3A%2F%2Ffediverse.tv%2Fw%2Fasdasdasda`
** In pleroma it goes to a url like: `https://barcelona.social/ostatus_subscribe?acct=https%3A%2F%2Ffediverse.tv%2Fw%2Fasdasdasd`
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):
```
51.91.68.227 - - [17/Feb/2022:19:26:24 +0100] "GET /w/ctfm1d6Aa4qxwZ7GNTp1VE HTTP/1.1" 200 6161 "-" "http.rb/5.0.4 (Mastodon/3.4.6; +https://mastodon.art/)"
51.15.244.243 - - [17/Feb/2022:19:23:21 +0100] "GET /w/ctfm1d6Aa4qxwZ7GNTp1VE HTTP/2.0" 200 6161 "-" "Pleroma 2.4.1; https://barcelona.social <admin@barcelona.social>"
5.9.52.23 - - [17/Feb/2022:19:29:21 +0100] "GET /w/ctfm1d6Aa4qxwZ7GNTp1VE HTTP/1.1" 200 6161 "-" "http.rb/5.0.4 (Mastodon/3.4.6; +https://mastodon.social/)"
```
The weird part abot the pleroma log is that I don't use that user, I use `drymer`.

**Comment by Booteille** [comment]
Hi.

I tried with the instance you linked and I got the same error than you.

But, I tried on `peertube.datagueule.tv` and `tube.distrilab.fr` and got expected behaviour.

On the instance `fediverse.tv` I couldn't access to the about page. It loads the same page as the one on which I actually am, which is not normal.

I think there are some troubles with your instances.

Did you try remote interaction with other instances than `fediverse.tv` to be sure we can replicate your issue on other PeerTube instances?

**Comment by drym3r** [comment]
I did not, I know it's probably about our instance, but not sure how to debug it. The "about page" issue is also weird, we've investigated and we're not able to find the source of the problem. If you could also give some directions about that, I'd appreciate it.

**Comment by drym3r** [comment]
Oh. Well I think that's weird, but it makes sense I guess. Maybe there should be some kind of advertising that says that non-listed videos won't work. I'll try it with a normal video today later.

**Comment by Booteille** [comment]
I don't know enough of how federation works to help you there. Probably it'll be better to wait for other to answer.

**Comment by Chocobozzz** [comment]
>  But I made the test video public and even reupload it, and couldn't comment on it.

Could you share public videos where remote interactions don't work?

**Comment by drym3r** [comment]
Sure, I just updated this video. https://fediverse.tv/w/1R2pnjC3oQ4pYTB9PSEjnK

**Comment by Chocobozzz** [comment]
> Sure, I just updated this video. https://fediverse.tv/w/1R2pnjC3oQ4pYTB9PSEjnK

It does work on my side. Could you details what are the errors on yours?

**Comment by Booteille** [comment]
I can confirm this works on my side too.

**Comment by drym3r** [comment]
Hum, I see it working too with mastodon.social, but it totally didn't when I post it. Does it make sense that it takes some time to federate or whatever?
Also, with barcelona.social doesn't work. Has anybody tried with pleroma?

**Comment by AnonymousWebHacker** [comment]
https://github.com/Chocobozzz/PeerTube/issues/4808

**Comment by drym3r** [comment]
Not that it's a big sample, but I asked to try to another person from another pleroma instance and it also doesn't work.

**Comment by squatica** [comment]
I just got the same problem between https://tilvids.com (peertube 3.4.1) and https://mstdn.io (mastodon 3.4). 

I can see the peertube tab made a request to https://mstdn.io/.well-known/webfinger?resource=acct:my-username@mstdn.io and it opens a new tab with mstdn.io, but just the normal homepage, no way to confirm interaction or write the comment. What should it do instead?

**Comment by drym3r** [comment]
@squatica I would open another issue for that. In my case, when the comments work, they can be seen in both parts, mastodon/pleroma server and peertube.

**Comment by drym3r** [comment]
I thought it may be pleroma versions, but I have both in 2.4.2 and 2.4.1 that work and don't so I'm pretty lost at this.

**Comment by Chocobozzz** [comment]
Hi,

I'm sorry but I'm a bit lost. @drym3r Could you summarize the issue?

**Comment by drym3r** [comment]
I think I'll just close it, because I believe is more related to pleroma itself than to peertube. My initial questions were answered, so thank you anyway!
